src/discussions/posts/post/PostFooter.test.jsx

```diff
@@ -81,4 +81,10 @@ describe('PostFooter', () => {
     renderComponent({ ...mockPost, groupId: 5, groupName: 'Test Cohort' });
     expect(screen.getByTestId('cohort-icon')).toBeTruthy();
   });
+  it('test follow button', async () => {
+    renderComponent(mockPost);
+    const followButton = screen.getByTestId("follow-button");
+    expect(followButton).toBeTruthy();
+    fireEvent.click(followButton);
+  });
```

Comment:
This test doesn't really do anything. The idea was to add a test to improve coverage, it didn't need to explicitly cover the follow button. However, if you want to test the follow button, it's better to test actual behaviour. 

Here is a test you can use for instance:
```suggestion
  it.each([[true, /unfollow/i], [false, /follow/i]])('test follow button when following=%s', async (following, message) => {
    renderComponent({ ...mockPost, following });
    const followButton = screen.getByRole('button', { name: /follow/i });
    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
    await act(async () => {
      fireEvent.mouseEnter(followButton);
    });
    expect(screen.getByRole('tooltip')).toHaveTextContent(message);
  });
```

With `it.each` we're parametrising the tests so that the same test can be used for following=true and false. 

You don't need a `data-testid` for the follow button because there is a better way to get to it using `screen.getByRole('button', {name: /follow/i});`. A testid is the last resort if you can't access the element some other way. 

We are then triggering the mouseover event we do this in an `act block` so that the UI state changes are fully settled before we make assertions. 

Finally we are looking for the tooltip that shows when you hover over the follow button. This tooltip will show the message "Follow" if you're not following the post, and "Unfollow" if you are following. We make an assertion based on that.

Reply:
@xitij2000 Thanks for the clarification. I updated the PR accordingly.